models/confirmation_test.go

```diff
@@ -229,6 +234,47 @@ func TestConfirmationContextCustomUnmarshaler(s *testing.T) {
 	})
 }
 
+func TestConfirmationCalculatesExpiresAt(t *testing.T) {
+	for cType := range Timeouts {
+		invite, err := NewConfirmation(cType, TemplateNamePasswordReset, USERID)
+		if err != nil {
+			t.Fatalf("expected nil, got %+v", err)
+		}
+		if invite.ExpiresAt == nil || invite.ExpiresAt.IsZero() {
+			t.Errorf("expected non-Zero ExiresAt")
+		}
+	}
+}
+
+func TestConfirmationIsExpired(t *testing.T) {
+	for cType := range Timeouts {
+		invite, err := NewConfirmation(cType, TemplateNameCareteamInvite, USERID)
+		if err != nil {
+			t.Fatalf("expected nil error, got %s", err)
+		}
+		if invite.IsExpired() {
+			t.Errorf("expected false, got true")
+		}
+		*invite.ExpiresAt = time.Unix(0, 0)
```

Comment:
This will fail if `invite.ExpiresAt` was previously `nil`. Consider:

```suggestion
        expiresAt := time.Unix(0, 0)
		invite.ExpiresAt = &expiresAt
```